src/diagrams/common/common.ts

```diff
@@ -0,0 +1,147 @@
+import DOMPurify from 'dompurify';
+import { MermaidConfig } from 'types/config';
+
+/**
+ * Gets the rows of lines in a string
+ *
+ * @param {string | undefined} s The string to check the lines for
+ * @returns {string[]} The rows in that string
+ */
+export const getRows = (s?: string): string[] => {
+  if (!s) return [''];
+  const str = breakToPlaceholder(s).replace(/\\n/g, '#br#');
+  return str.split('#br#');
+};
+
+/**
+ * Removes script tags from a text
+ *
+ * @param {string} txt The text to sanitize
+ * @returns {string} The safer text
+ */
+export const removeScript = (txt: string): string => {
+  return DOMPurify.sanitize(txt);
```

Comment:
@knsv The old code was throwing multiple github codescanner issues.
DOMPurify passed all tests, and we're able to remove a lot of custom logic.

<img width="609" alt="image" src="https://user-images.githubusercontent.com/10703445/185778260-4b2b454c-b081-4551-bed5-e2c18b18d9c6.png">

Reply:
Yes we are applying DOMPurify at the end now which makes this redundant.

Reply:
So, the previous logic was to remove script based on level.
<img width="532" alt="image" src="https://user-images.githubusercontent.com/10703445/186340071-9f848282-ddf4-497a-b6a4-417bf116e7bc.png">

But now, we just sanitize it regardless of level. Wouldn't that remove scripts the user actually wants?
<img width="839" alt="image" src="https://user-images.githubusercontent.com/10703445/186340189-0f04fa26-b46b-4500-93da-426a497dcd55.png">